sdk/src/client/secret/private_key.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Implementation of [`PrivateKeySecretManager`].
+
+use std::ops::Range;
+
+use async_trait::async_trait;
+use crypto::{
+    hashes::{blake2b::Blake2b256, Digest},
+    keys::bip44::Bip44,
+    signatures::{
+        ed25519,
+        secp256k1_ecdsa::{self, EvmAddress},
+    },
+};
+use zeroize::{Zeroize, Zeroizing};
+
+use super::{GenerateAddressOptions, SecretManage};
+use crate::{
+    client::{api::PreparedTransactionData, Error},
+    types::block::{
+        address::Ed25519Address, payload::transaction::TransactionPayload, signature::Ed25519Signature, unlock::Unlocks,
+    },
+};
+
+/// Secret manager based on a single private key.
+pub struct PrivateKeySecretManager(ed25519::SecretKey);
+
+impl std::fmt::Debug for PrivateKeySecretManager {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_tuple("PrivateKeySecretManager").finish()
+    }
+}
+
+#[async_trait]
+impl SecretManage for PrivateKeySecretManager {
+    type Error = Error;
+
+    async fn generate_ed25519_addresses(
+        &self,
+        _coin_type: u32,
+        _account_index: u32,
+        _address_indexes: Range<u32>,
+        _options: impl Into<Option<GenerateAddressOptions>> + Send,
+    ) -> Result<Vec<Ed25519Address>, Self::Error> {
+        let public_key = self.0.public_key().to_bytes();
+
+        // Hash the public key to get the address
+        let result = Blake2b256::digest(public_key).try_into().map_err(|_e| {
+            crate::client::Error::Blake2b256("hashing the public key while generating the address failed.")
+        })?;
+
+        crate::client::Result::Ok(vec![Ed25519Address::new(result)])
+    }
+
+    async fn generate_evm_addresses(
+        &self,
+        _coin_type: u32,
+        _account_index: u32,
+        _address_indexes: Range<u32>,
+        _options: impl Into<Option<GenerateAddressOptions>> + Send,
+    ) -> Result<Vec<EvmAddress>, Self::Error> {
+        panic!()
```

Comment:
There is no proper Error I can use and the enum is not non_exhaustive so I can't add a new variant, not sure what to do...

Reply:
What about `Err(Error::UnsupportedOperation.into())`?

Reply:
Changed